src/dataCorrectionsTool/incentiveEligibilityTool.js

```diff
@@ -0,0 +1,416 @@
+*/
+const humanReadableMDYTimeZoneOffset = (participantDate) => {
+    if (!participantDate) return 'N/A';
+    const date = new Date(participantDate);
+    const userTimezoneOffset = date.getTimezoneOffset() * 60000;
+    const adjustedDate = new Date(date.getTime() + userTimezoneOffset);
+    return adjustedDate.toLocaleDateString();
+};
+
+
+const setupModalContent = (participantPaymentRound) => {
+    const paymentRoundType = conceptIdToPaymentRoundMapping[participantPaymentRound];
+    const modalBody = document.querySelector('.modal-body');
+    if (!modalBody) return;
+    modalBody.textContent = `Are you sure you want to update the participant's incentive eligibility status for ${paymentRoundType}?`;
+};
+
+/**
+ * Check if participant is already incentive eligible (Flags)
+ * Async function to check if participant is already incentive eligible (Flags) and will return incentive eligilbility status and participant data
+ * @param {Object} participant - Participant object
+ * @param {String} selectedPaymentRound - Selected payment round  
+ * @returns {Object} - Response object { isEligibleForIncentive: Boolean, participantData: Object }
+*/
+const checkParticipantForEligibleIncentive = async (participant, selectedPaymentRound) => { 
+    const participantObj = participant;
+    const connectId = participantObj['Connect_ID'];
+
+    try {
+        const idToken = await getIdToken();
+        const response = await fetch(`${baseAPI}/dashboard?api=checkParticipantForEligibleIncentive&connectId=${connectId}&currentPaymentRound=${selectedPaymentRound}`, {
+            method: "GET",
+            headers: {
+                Authorization: "Bearer " + idToken,
+                "Content-Type": "application/json"
+            },
+        });
+        if (!response.ok) {
+            const error = (response.status + " Error" + ": " + (await response.json()).message);
+            throw new Error(error);
+        }
+
+        const responseObj = await response.json();
+        console.log("🚀 ~ checkParticipantForEligibleIncentive ~ responseObj:", responseObj)
+        return responseObj;
+    } catch (error) {
+        console.error("Failed to check partipant Eligibility: ", error);
+        throw error;
+    }
+};
+
+
+/**
+ * Update participant incentive eligibility ands returns updated data on success
+ * Async function to update participant incentive eligibility
+ * @param {Object} participant - Participant object
+ * @param {String} selectedPaymentRound - Selected payment round
+ * @param {String} selectedDateValue - Selected date of eligibility
+ * @returns {Promise<{
+ *   code: number,
+ *   data: Object,
+ *   message?: string
+ * }>}  Response object - { code: 200, data: {100767870,...} }
+*/
+const updateParticipantIncentiveEligibility = async (participant, selectedPaymentRound, selectedDateValue) => { 
+    const participantObj = participant;
+    const connectId = participantObj['Connect_ID'];
+
```

Comment:
I'm somewhat unclear on the purpose of having a separate participantObj variable here?